app/src/main/java/org/kiwix/kiwixmobile/webserver/wifi_hotspot/HotspotNotificationManager.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.webserver.wifi_hotspot
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import org.kiwix.kiwixmobile.R
+import org.kiwix.kiwixmobile.core.utils.HOTSPOT_SERVICE_CHANNEL_ID
+import org.kiwix.kiwixmobile.webserver.ZimHostActivity
+import javax.inject.Inject
+
+class HotspotNotificationManager @Inject constructor(
+  @get:Inject var notificationManager: NotificationManager,
+  private val context: Context
+) {
+
+  private fun hotspotNotificationChannel() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+      NotificationChannel(
+        HOTSPOT_SERVICE_CHANNEL_ID,
+        context.getString(R.string.hotspot_service_channel_name),
+        NotificationManager.IMPORTANCE_DEFAULT
+      ).apply {
+        description = context.getString(R.string.hotspot_channel_description)
+        setSound(null, null)
+        notificationManager.createNotificationChannel(this)
```

Comment:
this shouldn't be inside the apply, probably it should be outside it so you read what this method is actually doing naturally eg "the notification manager is creating a channel" and then if a reader is bothered they can read the details of that channel